crates/miden-tx/src/errors/mod.rs

```diff
@@ -27,6 +27,8 @@ use thiserror::Error;
 
 #[derive(Debug, Error)]
 pub enum NoteCheckerError {
+    #[error("invalid input note count {0})")]
+    InvalidInputNoteCount(u16),
```

Comment:
```suggestion
    #[error("input note count {0} is out of range)")]
    InputNoteCountOutOfRange(u16),
```

Nit: I'd use a name that is a bit more precise.

It would be nice to include the allowed range, but it seems pretty difficult to access the constant.